addons/mail/static/src/js/models/messages/message.js

```diff
@@ -59,6 +59,22 @@ var Message =  AbstractMessage.extend(Mixins.EventDispatcherMixin, ServicesMixin
     // Public
     //--------------------------------------------------------------------------
 
+    /**
+     * Get anonymous name of the message
+     *
+     * @override
+     * @return {string}
+     */
+    getAnonymousName: function () {
+        let originalChannelId = this.getOriginChannelID();
+        if (originalChannelId >= 0) {
+            let anonymousName = this.call('mail_service', 'getThread', originalChannelId).getAnonymousName();
```

Comment:
Maybe implement the default `_t("Anonymous")` value in the channel's `getAnonymousName` definition as well?

That way the behavior is similar if any other feature needs to get the anonymous name of the channel.
(It will also allow you to return the method result right away and simplify the code).